onnx/reference/op_run.py

```diff
@@ -403,6 +403,40 @@ def _run(self, *args, **kwargs):  # type: ignore
             f"Method '_run' must be overwritten for operator {self.__class__.__name__!r}."
         )
 
+    def _check_output_run(self, res: tuple[Any]) -> tuple[Any]:
+        """
+        Checks the output are from the expected type.
+        """
+        if not isinstance(res, tuple):
+            raise TypeError(
+                f"Method '_run' of class {self.__class__.__name__!r} does not return a tuple but {type(res)}."
+            )
+        if len(res) == 0:
+            raise ValueError(
+                f"Method '_run' of class {self.__class__.__name__!r} does not return any result."
+            )
+        if any(isinstance(t, tuple) for t in res):
+            dtypes = [type(t) for t in res]
+            raise TypeError(
+                f"One of the results returned by method '_run' of class {self.__class__.__name__!r} "
+                f"is a tuple, this is no onnx correponding type (Map, List, Tensor, SparseTensor). "
+                f"All returned types: {dtypes!r}."
+            )
+        res = tuple(  # type: ignore[assignment]
```

Comment:
Does the function also "fix" the outputs? Should it be named `_check_and_fix_outputs`? Or should we separate the invariance checking part from this fixing logic so the "check" function does not return values?

Reply:
In scikit-learn, they call that kind of function validate... but they do everything in one call. I prefer to have one call.